src/expressions/xquery/CompTextConstructor.ts

```diff
@@ -0,0 +1,36 @@
+import Expression, { RESULT_ORDERINGS } from '../Expression';
+import Specificity from '../Specificity';
+
+import castToType from '../dataTypes/castToType';
+import createNodeValue from '../dataTypes/createNodeValue';
+import sequenceFactory from '../dataTypes/sequenceFactory';
+
+class CompTextConstructor extends Expression {
```

Comment:
I suggest to remove the `Comp` prefix as we also do not use this for the other computed constructors.